File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/opensearch/src/main/java/com/example/search/ListDomainNames.java

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/
package com.example.search;

import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.opensearch.OpenSearchClient;
import software.amazon.awssdk.services.opensearch.model.DomainInfo;
import software.amazon.awssdk.services.opensearch.model.ListDomainNamesRequest;
import software.amazon.awssdk.services.opensearch.model.ListDomainNamesResponse;
import software.amazon.awssdk.services.opensearch.model.OpenSearchException;

import java.util.List;


public class ListDomainNames {

    public static void main(String[] args) {

        Region region = Region.US_EAST_1;
        OpenSearchClient searchClient = OpenSearchClient.builder()
                .region(region)
                .build();
        listAllDomains(searchClient);
        System.out.println("Done");
    }

    public static void listAllDomains(OpenSearchClient searchClient){

        try {
            ListDomainNamesRequest namesRequest = ListDomainNamesRequest.builder()
                    .engineType("OpenSearch")
                    .build();

            ListDomainNamesResponse response = searchClient.listDomainNames(namesRequest) ;
            List<DomainInfo> domainInfoList = response.domainNames();
            for (DomainInfo domain: domainInfoList)
                System.out.println("Domain name is "+domain.domainName());

        } catch (OpenSearchException e) {
            System.err.println(e.awsErrorDetails().errorMessage());
            System.exit(1);
        }
   }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/opensearch/src/main/kotlin/com/example/search/ListDomainNames.kt

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.example.search

import aws.sdk.kotlin.services.opensearch.OpenSearchClient
import aws.sdk.kotlin.services.opensearch.model.ListDomainNamesResponse
import aws.sdk.kotlin.services.opensearch.model.ListDomainNamesRequest

suspend fun main() {
    listAllDomains()
}

suspend fun listAllDomains() {

    OpenSearchClient { region = "us-east-1" }.use { searchClient ->
     val response: ListDomainNamesResponse = searchClient.listDomainNames(ListDomainNamesRequest {})
     response.domainNames?.forEach { domain ->
            println("Domain name is " + domain.domainName)
        }
    }
}
